lib/rails_autolink/helpers.rb

```diff
@@ -98,7 +98,7 @@ def auto_link_urls(text, html_options = {}, options = {})
                 href
               else
                 # don't include trailing punctuation character as part of the URL
-                while href.sub!(/[^#{WORD_PATTERN}\/-=&]$/, '')
+                while href.sub!(/[^#{WORD_PATTERN}\/-=&\-]$/, '')
```

Comment:
Can escape the hyphen that's already in the character class:
```suggestion
                while href.sub!(/[^#{WORD_PATTERN}\/\-=&]$/, '')
```